gemcalibration/src/common/Calibration.cc

```diff
@@ -0,0 +1,226 @@
+  *out << " { \"mon_state\":\"STOPPED\"}" << std::endl;
+*/
+}
+
+void gem::calib::Calibration::resumeAction(xgi::Input* in, xgi::Output* out)
+  throw (xgi::exception::Exception)
+{
+/*
+  CMSGEMOS_INFO("ShelfMonitor::startAction");
+  out->getHTTPResponseHeader().addHeader("Content-Type", "application/json");
+  //this->startMonitoring();
+  *out << " { \"mon_state\":\"RUNNING\"}" << std::endl;
+*/
+}
+
+void gem::calib::Calibration::pauseAction(xgi::Input* in, xgi::Output* out)
+  throw (xgi::exception::Exception)
+{
+  
+  CMSGEMOS_INFO("Calibration::pauseAction");
+  out->getHTTPResponseHeader().addHeader("Content-Type", "application/json");
+  //this->stopMonitoring();
+  *out << " { \"mon_state\":\"PAUSED\"}" << std::endl;
+  std::cout << "ciao! " << std::endl;
+}
+
+void gem::calib::Calibration::applyAction(xgi::Input* in, xgi::Output* out)
+  throw (xgi::exception::Exception)
+{
+    bool t_errorsOccured = false;
+    cgicc::Cgicc cgi(in);
+    for (auto it: m_scanParams.find(m_calType)->second){
+        it.second = cgi[it.first]->getIntegerValue();
+        CMSGEMOS_DEBUG("Calibration::applyAction : " << it.first << " = " << it.second);
+    }
+    
+    std::stringstream t_stream;
+    t_stream.clear();
+    t_stream.str(std::string());
+    for (unsigned int i = 0; i < NSHELF; ++i) {
+        t_stream.clear();
+        t_stream.str(std::string());
+        t_stream << "shelf"<< std::setfill('0') << std::setw(2) << i+1;
+        bool checked = false;
+        checked = cgi.queryCheckbox(t_stream.str());
+        if (checked) {
+            for (unsigned int j = 0; j < NAMC; ++j) { //SHELF.AMC
+                t_stream.clear();
+                t_stream.str(std::string());
+                t_stream << "shelf"<< std::setfill('0') << std::setw(2) << i+1 << ".amc" << std::setfill('0') << std::setw(2) << j+1;
+                checked = cgi.queryCheckbox(t_stream.str());
+                if (checked) {
+                    std::string amc_id = t_stream.str();
+                    amc_optical_links.emplace(amc_id, 0);
+                    t_stream << ".ohMask";
+                    uint32_t ohMask = std::stoul(cgi[t_stream.str()]->getValue(), 0, 16);
+                    if (ohMask > 0x3ff) {
+                        t_errorsOccured = true;
+                        CMSGEMOS_ERROR("Calibration::applyAction : OH mask for " << t_stream.str() << " is out of allowed boundaries! Ignoring it");
+                    } else{
+                        amc_optical_links.find(amc_id)->second = ohMask;
+                    }
+                } // end if checked for amc
+            } // end loop over NAMC 
+        } // end if checked for shelf
+    } //end loop over shelves
+    //TODO: should we check if at least one link is selected??
+    
+
+    //DACSCAN type
+    if (m_calType==DACSCANV3){
+      std::map<dacScanType_t, std::map<std::string, uint32_t> >::iterator it;
+      for(it=m_dacScanTypeParams.begin();it!=m_dacScanTypeParams.end();it++){
```

Comment:
* whitespace should follow the form `for (blah) {` (@lmoureaux has added many of my style requirements to the `CONTRIBUTING.md` guidelines)
* Use scoped range based loops with `auto &` or `auto const&` as necessary rather than declaring an iterator outside the loop (unless you have a valid reason not to)